Implement server

rianp opened the pull request:
## Changes Made:

1. Add EchoServer Class with a start server method
2. Add EchoServer test

## Acceptance Criteria

1. When I start my server it should connect to a given port.

[trello link](https://trello.com/c/IGqxEiTy)

Commit:
implement start server functionality

Commit:
change ci linter to check proper formatting

Commit:
change ci linter to check proper formatting

Commit:
reformat code to pass ci linter

Commit:
reformat code to pass ci linter

clairenguyen commented on app/src/main/java/echo/server/EchoServer.java:
```diff
@@ -0,0 +1,21 @@
+      System.out.println("Socket Created!");
+      return server_socket;
+    } catch (IOException e) {
+      throw new RuntimeException(e);
```
Wondering if you can add a test for this error case.

clairenguyen commented on app/src/test/java/echo/server/AppTest.java:
```diff
@@ -0,0 +1,16 @@
+class AppTest {
+  @Test
+  void appHasAGreeting() {
+    App classUnderTest = new App();
```
suggestion, you don't have to do this:  a common convention is to use `subject` as the variable name for the class under test

clairenguyen commented on app/src/test/java/echo/server/EchoServerTest.java:
```diff
@@ -0,0 +1,16 @@
+
+public class EchoServerTest {
+  @Test
+  void testServerSocket() throws IOException {
```
so unlike in python, we don't _have_ to use the word "test" at the front for our junit test :) Some people still do, but i feel like that's an extra word we that can just be taken out. the @test annotation describes that it's a test already imo. it might just be a personal preference though. maybe the other mentors have other opinions around this.

clairenguyen replied:
also, just FYI, i think it's common in junit to underscores in between certain groupings.  Here are some examples i found . it lists some "cons" for each naming convention and that might help you form an opinion on the style you prefer.

https://medium.com/@stefanovskyi/unit-test-naming-conventions-dd9208eadbea

rianp replied:
omg no more using test int he name! yay!!!!

clairenguyen commented on app/src/test/java/echo/server/EchoServerTest.java:
```diff
@@ -0,0 +1,16 @@
+  @Test
+  void testServerSocket() throws IOException {
+    ServerSocket serverSocket = EchoServer.createSocket();
+    assertNotNull(serverSocket, "Should create server socket when the server starts");
```
can we incorporate the string message as part of the test name? it provides a bit more description of what to expect in this test than just "test server socket"

clairenguyen reviewed (commented):
Yay! Super awesome that you got a basic server up and running pretty quickly! Good job! Left some comments about ways you could improve on tests.

ranizilpelwar commented on app/build.gradle.kts:
```diff
@@ -0,0 +1,41 @@
+/*
+ * This file was generated by the Gradle 'init' task.
```
Can you gitignore the bin files? See above - they are getting checked in

Commit:
add runtime exception error test

Commit:
move print functionality to console class

Commit:
linter refactors

linter refactors

linter refactors

linter refactor

linter refactor

linter refactor

clairenguyen commented:
@rianp have the comments above been resolved? if so, did you push up your changes to this PR?  can you click the "resolve conversation" button when you're done so we know that you're done working on them (even if there's no changes on your part)?

ranizilpelwar commented on app/src/test/java/echo/server/EchoServerTest.java:
```diff
@@ -0,0 +1,31 @@
+    Assertions.assertNotNull(serverSocket, "Should create server socket when the server starts");
+  }
+
+  @Test
```
Now that you have tests, it would be great to update the README. There are some assumptions in the current README:
- This assumes the user has Intellij and JDK installations already. For a user that wants to run your server locally, it is challenging to know what IDE to use or know how/where to install any dependencies. It would be nice to have this info here as well. 
- It would also be good to add a section that says how to run the tests, either in bash or in Intellij, with both being the best!
- BONUS: It would be awesome to show the user how to call the endpoint to try out the changes. Ex: curl command.

ranizilpelwar reviewed (approved):
Approved, assuming no pending changes from other mentor's comments.

Commit:
update readme
